source/file_layout.rst

```diff
@@ -42,25 +44,33 @@ All ASDF files must start with a short one-line header.  For example::
 
   #ASDF 0.1.0
 
-It is made up of the following parts (described in EBNF form)::
+It is made up of two parts, separated by white space characters:
+
+  - **ASDF token**: The constant string ``#ASDF``. This can be used to
+    quickly identify the file as an ASDF file by reading the first 5
+    bytes.  It begins with a ``#`` so it will be treated as a YAML
+    comment such that the :ref:`header` and the :ref:`tree` together
+    form a valid YAML file.
+
+  - **File format version**: The version of the low-level file format
+    that this file was written with.  This version may differ from the
+    version of the ASDF specification, and is only updated when a
+    change is made that affects the layout of file.  See
+    :ref:`versioning-conventions` for more information about these
+    versions.
 
-  asdf_token = "#ASDF"
-  major      = integer
-  minor      = integer
-  micro      = integer
-  header     = asdf_token " " major "." minor "." micro ["\r"] "\n"
```

Comment:
Maybe just keep
```
asdf_token = "#ASDF"
header       = asdf_token " " format_version ["\r"] "\n"
```
where the syntax for `format_version` is described in the semver spec?